File: birthday-cake-candles/birthday-cake-candles.go
package main

func birthdayCakeCandles(ns []int32) int32 {
	var max int32 = 0
	var freq int32 = 0

	for _, n := range ns {
		if n > max {
			max = n
			freq = 1
		} else if n == max {
			freq++
		}
	}

	return freq
}

File: birthday-cake-candles/BirthdayCakeCandles.java
import java.util.List;

public class BirthdayCakeCandles {

  public static int birthdayCakeCandles(List<Integer> ns) {
    int max = 0;
    int freq = 0;

    for (int i = 0; i < ns.size(); i++) {
      int n = ns.get(i);
      if (n > max) {
        max = n;
        freq = 1;
      } else if (n == max) {
        freq++;
      }
    }

    return freq;
  }

}
